### PR TITLE
Fix 307 part2

### LINKED ISSUE #307: integrate vxx's work
In the old bb repo, wvx was working on porting minor improvements from 1.13:
https://github.com/wvx/ja2-stracciatella/commits/20160205

I remember there were some problems with the warning work, but the featurettes mostly look fine. We need to decide what to accept at some point, hence this reminder.

MOVED:
- mod: Import your old IMP merc into a new game; related/tracked in #32 
- recheck and extract extender fix #624

IGNORED:
WIP: extend pathing (needs more work)
WIP: skip sleep explanation (needs more work)
WIP: vehicle inventory (probably needs more work)
externalise: perpetual crepitus queen corpse (cheating)
mod: recruit militia to squad (scope)
Inventory Management Extras (several things at once, some wip, some odd; too much work)
mod: soldiers can have all skill traits (get *all* traits if you meet stat criteria)
mod: accurate aim circle (unreliable also from my experience)
allow realtime tactical saving in ironman (it was optional, but contradictory to the difficulty setting's purpose)
Wdelete-non-virtual-dtor (unresolved leaks)
Warray (looks pointless)

### PR DESCRIPTION
**Pull request by lynxlynxlynx**
Fix #307 part2

### CODE REVIEW

File: Build/Tactical/Weapons.cc [hunk 1]
```diff
@@ -321,11 +318,14 @@ INT8 ArmourVersusExplosivesPercent( SOLDIERTYPE * pSoldier )
 
 static void AdjustImpactByHitLocation(INT32 iImpact, UINT8 ubHitLocation, INT32* piNewImpact, INT32* piImpactForCrits)
 {
+	UINT32 critical_damage_to_head = (GCM->getGamePolicy()->critical_damage_head_multiplier)*iImpact;
+//	UINT32 critical_damage_to_legs = (GCM->getGamePolicy()->critical_damage_legs_multiplier)*iImpact;
```
Comment (selaux) [inline Build/Tactical/Weapons.cc]: Could we add that as well? Seems like it was already started...

File: externalized/game.json [hunk 2]
```diff
@@ -74,6 +74,24 @@
...
+    //  vanilla_setting: 1, 99, 1.5
+    "chance_to_hit_minimum": 1,
+    "chance_to_hit_maximum": 99,
+    "tactical_head_damage_multiplier": 1.5,
+
+    //  IMP attribute point defaults
+    //  Zero attribute point bonus is what you receive for setting a skill to 0
+    //  Unspecified bonus is your initial pool of points to distribute
+    //  vanilla_setting: 85, 35, 15, 40
+    "IMP_MAX_ATTRIBUTE_POINTS": 85,
```
Comment (selaux) [inline externalized/game.json]: all other identifiers are all lowercase, we should also do that here. Also we could nest all imp related things into an "imp" object, to keep stuff more structured.

File: Build/Tactical/Turn_Based_Input.cc [hunk 3]
```diff
@@ -2212,6 +2213,14 @@ void GetKeyboardInput(UIEventKind* const puiNewEvent)
 			HandleShortCutExitState( );
     }
 
+		if ((InputEvent.usEvent == KEY_DOWN )&& ( InputEvent.usParam == 'l') && ( InputEvent.usKeyState & CTRL_DOWN ) && gGameOptions.ubGameSaveMode != DIF_DEAD_IS_DEAD)
```
Comment (selaux) [inline Build/Tactical/Turn_Based_Input.cc]: Nice Catch 👍

File: Build/Tactical/Weapons.cc (hunk 1 above)
Comment (lynxlynxlynx) [inline Build/Tactical/Weapons.cc]: Yes, it's trivial.

File: externalized/game.json (hunk 2 above)
Comment (lynxlynxlynx) [inline externalized/game.json]: good idea, I'll check it out.

**Comment by lynxlynxlynx** [comment]
That's it. :)

**Comment by selaux** [comment]
Thanks 👍

**Pull request merged**